### PR TITLE
Adjust nudging for cases where no enterable cell is found

### PR DESCRIPTION
**Pull request by hktk1f40**
- Randomly select one of the occupied "non-stupid" cells
- Notify blockers in that cell
- Wait until it can be entered
- Move to cell

This is related to #4736.

I still encountered one problem after the patch, if you are using cheats (instant build) the actor being deployed to the rally point cannot cannot notify the blocking actor which just moved out of the production facility. This is due to the condition that the blocking actor must be idle inside Mobile.OnNotifyBlockingMove().

### CODE REVIEW

**Comment by penev92** [comment]
That has been bothering me for a long time. Thanks.

**Comment by Mailaender** [comment]
:+1: I can confirm that this fixes #4736.

File: OpenRA.Mods.Common/Traits/Mobile.cs [hunk 1]
```diff
@@ -592,8 +591,30 @@ public void Nudge(Actor self, Actor nudger, bool force)
...
+					var move = new Move(self, cellInfo.Cell);
+					self.QueueActivity(Util.SequenceActivities(notifyBlocking, waitFor, move));
+
+					Log.Write("debug", "OnNudge (notify next blocking actor, wait and move) #{0} from {1} to {2}",
+						self.ActorID, self.Location, cellInfo.Cell);
+				}
+				else
+				{
+					Log.Write("debug", "OnNudge #{0} refuses at {1}",
+										self.ActorID, self.Location);
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Traits/Mobile.cs]: This doesn't look correct. (Maybe just combine it to two lines as well.)

**Comment by obrakmann** [comment]
This does indeed work nicely, thanks! :+1:

**Comment by obrakmann** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog/_compare/34b65b6310272c88a9306171e23979749746adaf%5E...34b65b6310272c88a9306171e23979749746adaf)

**Pull request merged**